contracts/swap/swap.go

```diff
@@ -89,11 +74,35 @@ type Params struct {
 	ContractCode    string
 	ContractAbi     string
 	ContractAddress common.Address
+	NetworkID       string
 }
 
 type simpleContract struct {
-	instance *contract.SimpleSwap
-	address  common.Address
+	instance  *contract.SimpleSwap
+	address   common.Address
+	networkID string
```

Comment:
The network id is the same for all contracts that a `Swap` instance will interact with and it's queried during `StartChequebook `. Why not save it one `Swap` instead of `simpleContract`?

Reply:
Makes sense!
Changing it